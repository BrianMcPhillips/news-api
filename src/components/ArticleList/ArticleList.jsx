import React from 'react';
import PropTypes from 'prop-types';
import Article from '../Article/Article';

const ArticleList = ({ articles }) => {
  const articleElements = articles.map(article => (
    <li key={article.url}>
      <Article {...article}/>
    </li>
  ));
  return (
    <ul>
      {articleElements}
    </ul>
  );
};

ArticleList.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    urlToImage: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  })).isRequired
};

export default ArticleList;
