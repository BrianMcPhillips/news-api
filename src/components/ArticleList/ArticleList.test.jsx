import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ArticleList from './ArticleList';

describe('ArticleList component', () => {
  afterEach(() => cleanup());

  it('renders an article list', () => {
    const { asFragment } = render(<ArticleList 
      articles={[
        {
          urlToImage: 'http://placeholder.com/villager',
          title: 'Best article ever',
          description: 'This is beyond a great article so insightful',
          author: 'test testerson',
          url: 'www.test.com'
        },
        {
          urlToImage: 'http://placeholder.com/villager',
          title: 'Second best article ever',
          description: 'This is a another great article so insightful',
          author: 'test testerson the third',
          url: 'www.test2.com'
        }
      ]}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
