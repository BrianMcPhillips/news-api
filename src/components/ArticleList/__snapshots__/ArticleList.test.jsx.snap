// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList component renders an article list 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <div>
        <img
          alt="Best article ever"
          src="http://placeholder.com/villager"
        />
        <h2>
          Best article ever
        </h2>
        <h4>
          This is beyond a great article so insightful
        </h4>
        <p>
          test testerson
        </p>
        <a
          href="www.test.com"
        >
          View Article
        </a>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Second best article ever"
          src="http://placeholder.com/villager"
        />
        <h2>
          Second best article ever
        </h2>
        <h4>
          This is a another great article so insightful
        </h4>
        <p>
          test testerson the third
        </p>
        <a
          href="www.test2.com"
        >
          View Article
        </a>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
